### DISCUSSION TITLE
How to Develop, Run, Debug from Source?

### DISCUSSION THREAD

**Discussion by sroca-81** [fedorovvl/tso_client#260] [General]
How to setup Visual Studio to run/compile source code? Or, alternatively, which IDE to use?

**Comment by fedorovvl** [comment]
i use Visual Studio 2019
download sources, open with VS and run build

**Reply by C-o-l-u-m-b-o** [reply]
Hi, and is there a way to debug the AIR / Javascript as well? 
I mean how to find how all those classes are named?
I mean names like this "globalFlash.gui.mFriendsList.GetFriendById(swmmo.application.mGameInterface.mCurrentViewedZoneID).username" or "event.data.buff.GetBuffDefinition().GetName_string()" 
Thanks :)

**Reply by fedorovvl** [reply]
u can open client.swf with ffdec or AS3 Sorcerer to inspect internal classes
for debugging u can use debug window in client

**Reply by sroca-81** [reply]
is possible to write a script running in background that show an alert for a specify event? is there an example or someone similar?
(sorry for bad english XD)

**Reply by fedorovvl** [reply]
depends on event.. for ex u can check 4-spechide.js for lines
```
var hideSpecTracker = game.getTracker('specHide', hideSpecHandler);
game.gi.channels.ZONE.addPropertyObserver("ZONE_REFRESHED", hideSpecTracker);
```
this is handler for zone refresh event.. there is several channels in game.gi.channels for different events

**Reply by sroca-81** [reply]
I try thanks :)
